libs/common/RespWriteUtils.cs

```diff
@@ -435,6 +436,49 @@ public static int GetIntegerAsBulkStringLength(int integer)
             return 1 + integerLenSize + 2 + sign + integerLen + 2;
         }
 
+        /// <summary>
+        /// Try to write a double-precision floating-point <paramref name="value"/> as bulk string.
+        /// </summary>
+        /// <returns><see langword="true"/> if the <paramref name="value"/> could be written to <paramref name="curr"/>; <see langword="false"/> otherwise.</returns>
+        [SkipLocalsInit]
+        public static bool TryWriteDoubleBulkString(double value, ref byte* curr, byte* end)
+        {
+            if (double.IsInfinity(value))
```

Comment:
We could add basic NaN handling here too if we want to. Nothing exercises these codepaths currently but this logic can later be shared to implement double serialization per RESP3 spec.

Reply:
Yes, that makes sense.

Reply:
This subnormal handling is not defined for RESP2 so this is just currently a mix of both RESP2 & RESP3. This is free to change in future, nothing _should_ be getting stored as -Inf, Inf or NaN